### PR TITLE
:bug:Binary-Artifacts: skip files that look like tests

### PR DESCRIPTION
**Pull request by evverx**
Should address https://github.com/ossf/scorecard/issues/1256

I'm not sure how to figure out whether files are tests or not
by just looking at their names but it seems almost all the tools
analyzing source code I've seen so far either skip or demote files
containing "test" in their names. Some go even farther and admit that
they can't detect tests and allow users to specify where tests are one
way or another

### CODE REVIEW

**Review by naveensrinivasan — APPROVED** [review]
Thanks!

File: checks/binary_artifact.go
Comment (azeemshaikh38) [inline checks/binary_artifact.go:101]: We have `isTestData` fn already which looks for test files - https://github.com/ossf/scorecard/blob/main/checks/file_utils.go#L51. Maybe use that here instead?

**Review by azeemshaikh38 — APPROVED** [review]

File: checks/binary_artifact.go
Comment (evverx) [reply checks/binary_artifact.go:101]: It can't be used directly because it looks for "testdata" files only. It's possible to relax it by letting it skip "test" files but I'm not sure it's a good idea in the sense that I wouldn't want to start filtering out more files than I should in other places just because of the Binary-Artifacts check. I'd characterize this PR as a stop gap that should probably even look like a stop gap :-)

Comment (azeemshaikh38) [reply checks/binary_artifact.go:101]: Makes sense. We can submit this as is for now.

**Comment by laurentsimon** [comment]
We discard binaries present in folders named `testdata` folder for this.

**Review by laurentsimon — COMMENTED** [review]
We discard binaries present in folders named `testdata` folder for this. There is also a question around whether discarding binaries is a feature for result producers (repo owners) or result consumers (anyone who want more info about a repo before using it). I agree it's useful for repo owners and we can add this as an ignore list option to scorecard (we can add expose it in the GitHub action's policy file). For consumers, the best way to solve this is for the community to agree on naming conventions: testdata is already used by certain (several ?) languages.

**Comment by evverx** [comment]
> There is also a question around whether discarding binaries is a feature for result producers (repo owners) or result consumers (anyone who want more info about a repo before using it).

Given that consumers usually report "issues" discovered by various code analyzers to maintainers I think they are all involved one way or another.

> For consumers, the best way to solve this is for the community to agree on naming conventions: testdata is already used by certain (several ?) languages.

I replied in https://github.com/ossf/scorecard/issues/1256#issuecomment-969232897

**Comment by laurentsimon** [comment]
is it fair to say this PR may be closed since we've merged https://github.com/ossf/scorecard/pull/1300?
I'm sure someone else will raise the same issue for tests. Shall we re-visit when this happens and/or when we have a better understanding of what's a test?

cc @david-a-wheeler would love to add `where test files are` in the yaml config you mentioned during last meeting!

**Comment by evverx** [comment]
> is it fair to say this PR may be closed since we've merged #1300?

I don't think it is. There are projects like elfutils that keep ELF files for testing purposes. Plus, basically almost all projects keeping their seed corpora close to their source code are affected I think

**Comment by evverx** [comment]
I think it can be closed because I don't think that just looking for `test` is enough though. I asked how CodeQL and LGTM do it in https://github.com/ossf/scorecard/issues/1074#issuecomment-972337872 so in the meantime it would probably make sense to just wait for a reply.

@laurentsimon on an unrelated note I'm not sure who attends those meetings and whether they can affect any roadmaps but if would be great if for example GitHub fixed Dependabot along with their security tab: https://github.com/ossf/scorecard/pull/1336. I understand that the idea is to probably discuss high-level issues but it would be great if they could also lead to visible improvements in the tools scorecard itself recommends.

**Comment by laurentsimon** [comment]
> I think it can be closed because I don't think that just looking for `test` is enough though. I asked how CodeQL and LGTM do it in [#1074 (comment)](https://github.com/ossf/scorecard/issues/1074#issuecomment-972337872) so in the meantime it would probably make sense to just wait for a reply.
> 
> @laurentsimon on an unrelated note I'm not sure who attends those meetings and whether they can affect any roadmaps but if would be great if for example GitHub fixed Dependabot along with their security tab: #1336. I understand that the idea is to probably discuss high-level issues but it would be great if they could also lead to visible improvements in the tools scorecard itself recommends.

let's create an issue to track other tools improvements, thanks for suggesting

**Pull request closed without merging**
